### PR TITLE
JSON round trip support for source chains

### LINKED ISSUE #61: serialize/deserialize for source chains

### PR DESCRIPTION
**Pull request by thedavidmeister**
fixes #61

### CODE REVIEW

File: core/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,13 @@
 #![deny(warnings)]
+
+#[macro_use]
+extern crate serde_derive;
+extern crate serde;
+extern crate serde_json;
+
 extern crate holochain_dna;
 extern crate wabt;
```
Comment (sphinxc0re) [inline core/src/lib.rs]: I would order the import this order:
```rust
extern crate foo;
extern crate bar;

mod blah;
mod blahblah;
mod blahblahblah;

use foo::Bar;
use foo::Bang;
use foo::Boom;
use foo::Bang;
```
**But** we have *external* crates and *internal* crates, the ones that are in the repo. SO I'm thinking of having the external ones above the internal one with a gap in between.

File: core/Cargo.toml [hunk 2]
```diff
@@ -9,4 +9,7 @@ holochain_agent = { path = "../agent" }
 chrono = "0.4"
 wasmi = { git = "https://github.com/paritytech/wasmi.git"}
 wabt = "0.2.0"
-snowflake = "1.2"
\ No newline at end of file
+snowflake = "1.2"
+serde = "1.0"
+serde_derive = "1.0"
+serde_json = { version = "1.0", features = ["preserve_order"] }
```
Comment (sphinxc0re) [inline core/Cargo.toml]: Wouldn't it be better to change the versions to just `"1"`?

Comment (thedavidmeister) [reply core/Cargo.toml]: well wouldn't that unpin the version and allow the minor version to float?

File: core/src/lib.rs (hunk 1 above)
Comment (thedavidmeister) [reply core/src/lib.rs]: what are you wanting me to change here?

Comment (sphinxc0re) [reply core/src/lib.rs]: For example, `holochain_dna` is *internal*, so I would move it below the *external* crates with an empty line between

File: core/Cargo.toml (hunk 2 above)
Comment (thedavidmeister) [reply core/Cargo.toml]: this was copied from elsewhere in the repo, i think this convo lives here: https://github.com/holochain/holochain-rust/issues/48

**Pull request merged**